code/mobs/mob.dm

```diff
@@ -73,12 +86,54 @@
 		return TRUE
 	src << "You do not have space for that."
 
+/mob/proc/equip(obj/item/equipable/item)
+	if(!can_equip(item))//already thinking about adding a text that tells you why you couldnt equip
+		return
+	//okay basic overlays working
+	equipment[item.item_slot] = item
+	item.on_equip(src)
+	update_stats()
+	update_appearance()
+	usr << "equipped: [item]"
+
+/mob/proc/take_off(item_slot)
+	var/obj/item/equipable/in_slot = equipment[item_slot]
+	if(!can_take_off(in_slot))
+		return FALSE
+	equipment.Remove(item_slot)
+	update_appearance()
+	usr << "taken off: [in_slot]"
+
+/mob/proc/can_equip(obj/item/equipable/I)
+	if(isnull(equipment[I.item_slot]))
+		return TRUE
+	else
+		return FALSE
+
+/mob/proc/can_take_off(obj/item/equipable/I)
+	if(!I.cursed)
+		return TRUE
+	else
+		return FALSE
+
+//for now just going to handle the overlays
+/mob/proc/update_appearance()
+	var/list/overlays_list = list()
+	if((equipment.len == 0))
+		update_overlays(overlays_list)
+		return
+
+	for(var/I in equipment)
+		overlays_list.Add(equipment[I].create_overlay())
+	update_overlays(overlays_list)
```

Comment:
see the `/atom/proc/update_overlays` comment, updating overlays should be done in `update_overlays` not in `update_appearance`, this proc should only call other procs that update different parts of the appearance.